synapse/lib/types.py

```diff
@@ -481,6 +500,42 @@ def indxByEq(self, valu):
 
         return Type.indxByEq(self, valu)
 
+
+class Range(Type):
+
+    _opt_defs = {
+        ('subtype', None),
+    }
+
+    def postTypeInit(self):
+        self.subtype = self.opts.get('subtype')
```

Comment:
lets have his subtype opt be a typedef (name, {opts}) and construct the instance here using model.type(tdef[0]).clone(tdef[1]) so all subsequent uses can just have the type object directly.  Also, lets just call the option "type" :)

Reply:
fixed